
# **BIOR93 Applied Bioinformatics - Phylogenetics Module**

Teachers Jadranka Rota and Etka Yapar

## **Introduction**

In this course we will learn in practice how to download data from GenBank, create a multigene dataset, analyse the dataset using maximum likelihood and Bayesian methods, as well as multispecies coalescent methods, and how to manipulate trees for publication.

## **Program**

**Location:** Computer lab (D226), Biologihuset, Lund University, SÃ¶lvegatan 35, 223 62 Lund


| Hours\Days  | May 22                   | May 23                                          | May 24                                                    |
|-------------|--------------------------|-------------------------------------------------|-----------------------------------------------------------|
| 9:00-10:00  | ----                     | Introduction to molecular phylogenetics              |  Properties of molecular data |
| 10:00-12:00 | ----                     | Tutorials 3 and 4                               | Tutorial 6                                                |
| 12:00-13:00 | Lunch                    | Lunch                                           | Lunch                                                     |
| 13:00-14:00 | Introduction to Module 5 | Introduction to model-based phylogenetics | Tutorial 6 (cont.)                                        |
| 14:00-17:00 | Data formats and Tutorials 1 and 2        | Tutorial 5                                      | Tutorial 7                                                |

<!--  -->




## **Index**

### [Course slides](./Lectures/)

Here we will upload all the files and information shared during the lectures!


### [Data used in the course](./Data/)

Here you will find the files we create during the course. (If at some point you are at a loss because your attempts to produce the results for the next step failed, here you can find the files you will need so you can keep up with the class.)


### [Tutorials](./Tutorials/)

Here are deposited the instructions for the assignments and exercises.


 (1) [Learn to generate datasets](./Tutorials/DatasetManipulation/)
	
 (2) [Align the gene markers](./Tutorials/Alignments/)
 
 (3) [Select the best evolutionary model and partitioning scheme](./Tutorials/ModelSelection/)
 
 (4) [Maximum likelihood methods](./Tutorials/MaximumLikelihood/) 
 
 (5) [Bayesian methods](./Tutorials/BayesianInference/)
 
 (6) [Gene tree/species tree analysis](./Tutorials/ASTRAL/)
 
 (7) [Tree manipulation](./Tutorials/TreeManipulation/)
 
 (8) [Assignment](https://docs.google.com/spreadsheets/d/1rFkAeei_6dFn8auxJxbzCpq7qABZjHXpByE32seXyQQ/edit?usp=sharing)
 


### [Programs used in the tutorials](./Software/)

Here you can find a list of the programs we are going to use and links to where you can download them.
 
 * **Text Editor:**: We will use Visual Studio Code

 * [MAFFT](https://mafft.cbrc.jp/alignment/software/)

 * [AliView](http://www.ormbunkar.se/aliview/downloads/)

 * [IQ-TREE2](http://www.iqtree.org/)
 
 * [MrBayes](http://nbisweden.github.io/MrBayes/)
	
 * [FigTree](http://tree.bio.ed.ac.uk/software/figtree/)

 * [Tracer](https://github.com/beast-dev/tracer/releases/tag/v1.7.1)
 
 * [ASTRAL](https://github.com/smirarab/ASTRAL)

